gdk/commands/component/PublishCommand.py

```diff
@@ -175,7 +191,11 @@ def get_next_version(self):
         logging.debug("Fetching private components from the account.")
         try:
             c_name = self.project_config["component_name"]
-            c_next_patch_version = self.greengrass_client.get_highest_component_version_()
+            region = self.project_config["region"]
+            account_num = self.project_config["account_number"]
+            component_arn = f"arn:aws:greengrass:{region}:{account_num}:components:{c_name}"
```

Comment:
partitions.

Reply:
Can you elaborate? :)

Reply:
Saranya knows :)

`aws` is only valid for most regions, but not all. cn-north-1 for example needs `aws-cn` instead of `aws`.

Reply:
Ahhh, I see. Thanks for the info 👍

Reply:
[More info on ARNs.](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html#:~:text=type%3Aresource-id-,partition,-The%20partition%20in).